### DISCUSSION TITLE
`context.res` is an empty object when using `getServerSideProps` and `experimental-edge` runtime

### DISCUSSION THREAD

**Discussion by remorses** [vercel/next.js#41743] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
   Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.6.0: Mon Aug 22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64
    Binaries:
      Node: 16.16.0
      npm: 8.14.0
      Yarn: 1.22.17
      pnpm: 7.9.5
    Relevant packages:
      next: 12.3.2-canary.38
      eslint-config-next: 12.3.0
      react: 18.2.0
      react-dom: 18.2.0
```

### Describe the Bug

 `getServerSideProps` `context.res` is an empty object when using `experimental-edge` runtime

The cause of this problem:
https://github.com/vercel/next.js/blob/20f5db06fa5d2b627af76ed84a9f317c21e3ab04/packages/next/server/web-server.ts#L375

### Expected Behavior

`context.res` should at least have some basic methods like `setHeader` or just be a `NextResponse`

### Link to reproduction

https://github.com/remorses/try-next-getserversideprops-edge

### To Reproduce

Run `next dev` and see that `res` has no methods

**Comment by cheesycod** [comment]
Same, can someone do something about it
